functorch/csrc/BatchRulesViews.cpp

```diff
@@ -499,6 +510,7 @@ TORCH_LIBRARY_IMPL(aten, FT_BATCHED_KEY, m) {
   m.impl("flatten.using_ints", static_cast<decltype(&ATEN_FN2(flatten, using_ints))>(native::flatten));
   VMAP_SUPPORT(flip, flip_batch_rule);
   m.impl("trace", trace_decomp);
+  m.impl("trace_backward", trace_backward_decomp);
```

Comment:
I think we might need to register the decomposition over at https://github.com/pytorch/functorch/blob/main/functorch/csrc/PyTorchOperatorHacks.cpp#L192 .

Our composition tests don't actually cover the following:
```
import torch
from functorch import grad, vmap

def sum_grad_trace(x):
  return grad(torch.trace)(x).sum()

x = torch.randn(2, 2, 2)
vmap(grad(sum_grad_trace))(x)
```
does this pass under your PR?

Reply:
Unfortunately, no, this does not pass